### ISSUE TITLE
Add support for command line executables in app-specific settings [enhancement]

### ISSUE THREAD

**Issue by kode54** [noah-nuebling/mac-mouse-fix#55]
## Description
Basically, allow adding any arbitrary file to the app-specific settings, so long as, on disk, it has the  x attribute for at least the running user. Not folders, but executable files.

## Rationale
I would like to block Mouse Fix for the version of the java runtime which is retrieved by the Minecraft Launcher. Unfortunately, it doesn't appear to run as an app bundle, but rather a command line executable embedded in a bundle, located under a directory inside ~/Library/Application Support/minecraft/. This folder will change any time the Runtime is updated, but at least being able to add the current version would be handy.

Even easier would be if it were possible to drag a running application's dock icon onto the app-specific settings page. Currently, I can only drag the `java` binary which I locate with Finder, and because it's not an .app bundle, the whole PrefPane crashes itself.

## Additional info
For now, I shall just turn off the Mouse Fix globally while playing the game. Incidentally, with near identical settings, including resolution, Optifine, and Sildur's shaders, Extreme with Volumetric Lighting, it runs much better on macOS than Windows on the same machine. Damn, AMD's OpenGL Windows drivers suck.

**Comment by noah-nuebling** [comment]
Hey there @kode54, thanks for your detailed feedback!

This is a really good idea, which I hadn't considered. 
This app currently uses bundle identifiers to handle app-specific overrides.
Changing this to be based around executable paths might quite cumbersome and needs some consideration.

One issue I can think of is that Mac Mouse Fix has preinstalled overrides for apps that are known to cause issues with scrolling. This will be hard to implement well, if we base overrides around file paths of executables I recon.
The best solution might be to allow for both file path based overrides as well as bundle based ones but that should be even harder to do.

I probably won't get around to fixing this very soon unfortunately, but it's definitely on the agenda.

---

Drag and drop from the Dock would be super neat, but the Dock doesn't support Drag and Drop in that way as far as I know  (Because dragging Apps out of the Dock is what you do to remove them)

That's crazy that it runs better on macOS, first game I've ever heard of doing that!

**Comment by kode54** [comment]
Bumping this so the damn stale bot doesn't close it.

Never mind, looks like you just close issues after 6 months regardless of activity. May as well save your bot the trouble, then.

**Comment by kode54** [comment]
Yeah, sorry about that, I have been known to get annoyed by stale bots. Will keep this one open as the reference for that issue.

**Comment by noah-nuebling** [comment]
No worries, thank you!
